cordapp/src/main/kotlin/com/r3/corda/finance/obligation/client/flows/UpdateSettlementMethod.kt

```diff
@@ -10,66 +10,81 @@ import com.r3.corda.finance.obligation.types.SettlementMethod
 
-        // 2. This flow should only be started by the beneficiary.
-        val obligee = obligation.withWellKnownIdentities(resolver).obligee
-        check(ourIdentity == obligee) { "This flow can only be started by the obligee. " }
+            // 3. Add settlement instructions.
+            progressTracker.currentStep = UPDATING
+            val obligationWithSettlementTerms = obligation.withSettlementMethod(settlementMethod)
 
-        // 3. Add settlement instructions.
-        progressTracker.currentStep = UPDATING
-        val obligationWithSettlementTerms = obligation.withSettlementMethod(settlementMethod)
+            // 4. Build transaction which adds settlement terms.
+            progressTracker.currentStep = BUILDING
+            val signingKey = listOf(obligationWithSettlementTerms.obligee.owningKey)
+            val notary = serviceHub.networkMapCache.notaryIdentities.firstOrNull()
+                    ?: throw FlowException("No available notary.")
+            val utx = TransactionBuilder(notary = notary).apply {
+                addInputState(obligationStateAndRef)
+                addOutputState(obligationWithSettlementTerms, ObligationContract.CONTRACT_REF)
+                addCommand(ObligationCommands.UpdateSettlementMethod(), signingKey)
+            }
 
-        // 4. Build transaction which adds settlement terms.
-        progressTracker.currentStep = BUILDING
-        val signingKey = listOf(obligationWithSettlementTerms.obligee.owningKey)
-        val notary = serviceHub.networkMapCache.notaryIdentities.firstOrNull()
-                ?: throw FlowException("No available notary.")
-        val utx = TransactionBuilder(notary = notary).apply {
-            addInputState(obligationStateAndRef)
-            addOutputState(obligationWithSettlementTerms, ObligationContract.CONTRACT_REF)
-            addCommand(ObligationCommands.UpdateSettlementMethod(), signingKey)
-        }
+            // 5. Sign transaction.
+            progressTracker.currentStep = SIGNING
+            val stx = serviceHub.signInitialTransaction(utx, signingKey)
 
-        // 5. Sign transaction.
-        progressTracker.currentStep = SIGNING
-        val stx = serviceHub.signInitialTransaction(utx, signingKey)
+            // 6. Finalise transaction and send to participants.
+            progressTracker.currentStep = FINALISING
+            val obligorSession = initiateFlow(obligor as Party)
+            return subFlow(FinalityFlow(stx, setOf(obligorSession), FINALISING.childProgressTracker())).tx
+        }
+    }
 
-        // 6. Finalise transaction and send to participants.
-        progressTracker.currentStep = FINALISING
-        return subFlow(FinalityFlow(stx, FINALISING.childProgressTracker())).tx
+    @InitiatedBy(Initiator::class)
+    class Responder(val otherFlow: FlowSession) : FlowLogic<WireTransaction>() {
+        @Suspendable
+        override fun call(): WireTransaction {
```

Comment:
I usually return the signed transaction here, so you have the signatures, if you need them.

Reply:
However, here i  think i returned wire transaction so that it was rendered in the shell (which was a bit of a hack) :)